src/Operation/Find.php

```diff
@@ -293,33 +293,49 @@ public function execute(Server $server)
         return $cursor;
     }
 
+    public function getCommandDocument()
+    {
+        return $this->createCommandDocument();
+    }
+
     /**
-     * Create options for executing the command.
-     *
-     * @see http://php.net/manual/en/mongodb-driver-server.executequery.php
-     * @return array
+     * Construct a command document for Find
      */
-    private function createOptions()
+    private function createCommandDocument()
     {
-        $options = [];
+        $cmd = ['find' => $this->collectionName, 'filter' => new BSONDocument($this->filter)];
```

Comment:
I assume the intention here is to ensure that an array like `[]` still encodes as a BSON document. We can shortcut this with an `(object)` cast, which is exactly what [`BSONDocument::bsonSerialize()`](https://github.com/mongodb/mongo-php-library/blob/1.3.0/src/Model/BSONDocument.php#L72) would do (after copying `$this->filter` a couple of times and creating an ArrayObject instance).